### DISCUSSION TITLE
Error handling in a scenario

### DISCUSSION THREAD

**Discussion by tarikcarli** [TouK/nussknacker#3801] [Q&A]
Hello everybody,
I would like to know why we did not implement an error-handling path in a scenario like try-catch blocks. For example, we can store messages that cause errors in other Kafka topics and we consume them later with other scenarios. I am curious about why this feature is missing or do I miss something that actually is not that important. 

Thanks for reading, I look forward to seeing your responses.

**Comment by arkadius** [comment]
Hello @tarikcarli . Actually exact mechanism as you described is available in Nussknacker. It is done by [exceptionHandler](https://nussknacker.io/documentation/docs/integration/KafkaIntegration#exception-handling). So far we haven't tested it in the setup that another scenario consumes errors from the previous one. Probable some additional step is required like specifying json schema for error topic.

Which distribution are you using? docker / quickstart / helmchart / binary?
Which engine? Flink or Lite?

**Comment by tarikcarli** [comment]
@arkadius Thanks for your response. I really appreciate. This is the one that exactly what I need.
I use Nussknacker with docker setup. I configured quickstart project with my needs and I run scenarios into Flink engine. I run Nussknacker on wsl on windows, windows, and ubuntu successfully.

**Reply by arkadius** [reply]
Thanks for feedback. Let me know if I can help you with other issues.
